aspnetcore/tutorials/min-web-api.md

```diff
@@ -0,0 +1,384 @@
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+[!INCLUDE[](~/includes/trustCertVSC.md)]
+
+Press Ctrl+F5 to run the app. In a browser, go to following URL: [https://localhost:5001](https://localhost:5001)
+
+<!-- add VS Mac later 
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+Select **Run** > **Start Debugging** to launch the app. Visual Studio for Mac launches a browser and navigates to `https://localhost:<port>`, where `<port>` is a randomly chosen port number. 
+-->
+
+---
+
+`Hello World!` is displayed in the browser.
+
+## Add NuGet packages
+
+NuGet packages must be added to support the database and diagnostics used in this tutorial.
+
+# [Visual Studio](#tab/visual-studio)
+
+* From the **Tools** menu, select **NuGet Package Manager > Manage NuGet Packages for Solution**.
+* Select the **Browse** tab, and verify that **Include prerelease** is checked.
+* Enter **Microsoft.EntityFrameworkCore.InMemory** in the search box, and then select `Microsoft.EntityFrameworkCore.InMemory`.
+* Select the **Project** checkbox in the right pane and then select **Install**.
+* Follow the preceding instructions to add the `Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore` package.
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+* Run the following commands:
+
+   ```dotnetcli
+   dotnet add package Microsoft.EntityFrameworkCore.InMemory --prerelease
+   dotnet add package Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore --prerelease
+     ```
+
+<!-- add VS Mac later 
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+Open a command terminal in the project folder and run the following command:
+
+   ```dotnetcli
+   dotnet add package Microsoft.EntityFrameworkCore.InMemory
+   ```
+   -->
+---
+
+## Add the API code
+
+Replace the contents of the *program.cs* file with the following sample app code:
+
+[!code-csharp[](min-web-api/samples/6.x/todo/Program.cs?name=snippet_all)]
+
+## The model and database context classes
+
+The sample app contains the following model:
+
+[!code-csharp[](min-web-api/samples/6.x/todo/Program.cs?name=snippet_model)]
+
+A *model* is a set of classes that represent the data that the app manages. The model for this app is the `Todo` class.
+
+The sample app also contains the following database context class:
+
+[!code-csharp[](min-web-api/samples/6.x/todo/Program.cs?name=snippet_cntx)]
+
+The *database context* is the main class that coordinates [Entity Framework](/ef/core/) functionality for a data model. This class is created by deriving from the <xref:Microsoft.EntityFrameworkCore.DbContext?displayProperty=fullName> class.
+
+The following highlighted code adds the database context to the [dependency injection (DI)](xref:fundamentals/dependency-injection) container:
+
+[!code-csharp[](min-web-api/samples/6.x/todo/Program.cs?name=snippet_DI&highlight=3)]
+
+ The DI container provides access to the database context and other services.
+
+## Install Postman to test the app
```

Comment:
We have http-repl in the box...shouldn't we use our own tools here to encourage test?

Reply:
Agreed, I'll make that change.

Reply:
I think we should use Postman, it's a far nicer experience. We can mention http-repl if you like or perhaps use it for the first request, but Postman is just far superior.

Reply:
> I think we should use Postman, it's a far nicer experience. We can mention http-repl if you like or perhaps use it for the first request, but Postman is just far superior.

I never disagree with the boss, but we've found http-repl much easier to use than postman.  Most of the issues we have on WebAPI are related to Postman. The klugy security workaround isn't the biggest problem. Can we compromise?  Show  both but omit most of the Postman images (that change every 6 months) and just have postman instructions.